integration/container_stats_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+const (
+	defaultImage = "gcr.io/google_containers/pause:3.0"
+)
+
+func TestContainerStats(t *testing.T) {
+	t.Logf("Create a pod config and run sandbox container")
+	sbConfig := PodSandboxConfig("sandbox1", "stats")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sb))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sb))
+	}()
+	t.Logf("Fetch stats for sandbox container")
+	s, err := runtimeService.ContainerStats(sb)
+	t.Logf("Verify stats received for sandbox container")
+	require.NoError(t, err)
+	testStats(t, s)
+}
+
+func TestContainerListStats(t *testing.T) {
+	t.Logf("Create a pod config and run sandbox container")
+	sbConfig := PodSandboxConfig("running-pod", "statsls")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sb))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sb))
+	}()
+	t.Logf("Create a container config and run container in a pod")
+	containerConfig := ContainerConfig("container1", defaultImage)
+	cn, err := runtimeService.CreateContainer(sb, containerConfig, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cn))
+	}()
+	require.NoError(t, runtimeService.StartContainer(cn))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cn, 10))
+	}()
+	t.Logf("Fetch all container stats")
+	stats, err := runtimeService.ListContainerStats(&runtime.ContainerStatsFilter{})
+	require.NoError(t, err)
+	require.NotEmpty(t, stats)
+	t.Logf("Verify all container stats")
+	for _, s := range stats {
+		testStats(t, s)
+	}
+}
+
+func testStats(t *testing.T, s *runtime.ContainerStats) {
+	require.NotEmpty(t, s.Attributes)
```

Comment:
nit: Use `Getter`, which will save many lines. :)

`Getter` already handles the case that the original object is nil, it will just return zero value.